packages/wallets/src/wallets/evm.ts

```diff
@@ -123,4 +123,47 @@ export class EVMWallet extends Wallet<EVMChain> {
             chain: toViemChain(this.chain),
         });
     }
+
+    private async createTransaction(transaction: FormattedEVMTransaction): Promise<CreateTransactionSuccessResponse> {
+        const transactionCreationResponse = await this.apiClient.createTransaction(this.walletLocator, {
+            params: {
+                signer: this.signer.locator(),
+                chain: this.chain,
+                calls: [transaction],
+            },
+        });
+        if ("error" in transactionCreationResponse) {
+            throw new TransactionNotCreatedError(JSON.stringify(transactionCreationResponse));
+        }
+
+        return transactionCreationResponse;
+    }
+
+    private buildTransaction(params: EVMTransactionInput): FormattedEVMTransaction {
+        if ("transaction" in params) {
+            return { transaction: params.transaction };
+        }
+
+        if (params.abi == null) {
```

Comment:
If abi is null, shouldn't we support that it should either have the value or they are passing in the data directly?